### PR TITLE
Provide Stop func to ImageGCManager

### PR DESCRIPTION
**Pull request by tedyu**
**What type of PR is this?**
/kind cleanup

This fixes issue #77227

**What this PR does / why we need it**:
This PR adds Stop func to ImageGCManager so that the manager can be stopped at time of shutdown.

```release-note
NONE
```

### CODE REVIEW

**Review by mattjmcnaughton — APPROVED** [review]
/lgtm
/assign @tallclair

**Comment by tedyu** [comment]
ping @dchen1107 @yujuhong for approval

File: pkg/kubelet/images/image_gc_manager.go [hunk 1]
```diff
@@ -54,6 +54,9 @@ type ImageGCManager interface {
 	// Start async garbage collection of images.
 	Start()
 
+	// Stop async garbage collection of images.
```
Comment (smarterclayton) [inline pkg/kubelet/images/image_gc_manager.go]: No one calls stop, why would you need this?

If you plan on adding stop, do so in this commit.

/hold

Comment (tedyu) [reply pkg/kubelet/images/image_gc_manager.go]: I added the call to Stop()

Please take another look.

**Comment by tedyu** [comment]
Tests passed.

@smarterclayton 
Can you take a look ?

thanks

**Comment by tedyu** [comment]
gentle ping @smarterclayton

**Comment by tedyu** [comment]
gentle ping @smarterclayton

**Comment by tedyu** [comment]
ping @smarterclayton

**Comment by tedyu** [comment]
gentle ping @smarterclayton

**Comment by tedyu** [comment]
@smarterclayton 
Can the /hold be lifted ?

**Comment by tedyu** [comment]
@smarterclayton
Can the /hold be lifted ?

Thanks

**Comment by tedyu** [comment]
@smarterclayton
Can the /hold be lifted ?

**Comment by tedyu** [comment]
ping @smarterclayton for review

**Comment by yujuhong** [comment]
Who's going to call `ImageManager.Stop()` and when? Do we actually have a plan of supporting clean shutdown for kubelet? 

/cc @dashpole

**Comment by tedyu** [comment]
The Stop func is called by Kubelet#initializeModules when kl.oomWatcher.Start() fails.

**Comment by dashpole** [comment]
> The Stop func is called by Kubelet#initializeModules when kl.oomWatcher.Start() fails.

But it also returns an error there, which causes the kubelet to fail anyways.  It should be safe to use wait.NeverStop here, unless there is "cleanup" work we want to do in the image gc manager when we get the signal.  Additionally, if the kubelet stops for other reasons other than the OOM manager not starting we won't get the stop signal.

**Comment by tedyu** [comment]
For image gc manager, there is imageCache . After image gc manager stops, we can empty images field of imageCache.
```
other reasons other than the OOM manager not starting we won't get the stop signal.
```
oomWatcher is the one which returns error from Start().
Other managers don't return error from Start().

I can go through other managers but wonder if that is in scope of this PR.

**Comment by dashpole** [comment]
> After image gc manager stops, we can empty images field of imageCache.

When the kubelet process dies, the go runtime should handle freeing memory used by data structures in all goroutines.  Generally, the stop pattern is useful for ending external connections, such as finishing writes to open files, or closing open connections gracefully.  But we don't do any of that AFAICT, so I don't think this is necessary.

**Comment by tedyu** [comment]
We can revisit when there is ever a need.

**Pull request closed without merging**